icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6742,7 +6742,7 @@ static void checkTerminating(const char* locale, const char* inLocale)
     status = U_ZERO_ERROR;
     length = uloc_getDisplayName(
         locale, inLocale, buff, preflight_length, &status);
-    u_austrcpy(cbuff, buff);
+    u_austrncpy(cbuff, buff, length);
```

Comment:
The other problem with all of these is that the test code is reading from `buff` even when `U_FAILURE(status)` which means that there may be nothing but garbage in the buffer. It would be better to do something like

```
const char *cbuff = U_SUCCESS(status) ? aescstrdup(buff, length) : "(undefined when failure)";
```